From af82ad9159e471dd215cf1b2bcaa4209a8049ec8 Mon Sep 17 00:00:00 2001
From: Hariprasad Kelam <hkelam@marvell.com>
Date: Tue, 3 Nov 2020 16:01:19 +0530
Subject: [PATCH 0921/1921] octeontx2-af: sync kernel structures with firmware

cgx_link_stats structure fields are updated in firmware to
support lmac_type and link_mode. unsupported port field is
replaced with lmac_type. update kernel structure as well to
sync with firmware

Change-Id: I3f6cd05a97c4f2c4cc1bd19d8cd7433f51806076
Signed-off-by: Hariprasad Kelam <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/39254
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c       | 1 -
 drivers/net/ethernet/marvell/octeontx2/af/cgx_fw_if.h | 7 ++++---
 drivers/net/ethernet/marvell/octeontx2/af/mbox.h      | 1 -
 3 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 21d95b899b8a..b338906c1dd9 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -1190,7 +1190,6 @@ static inline void link_status_user_format(u64 lstat,
 	linfo->speed = cgx_speed_mbps[FIELD_GET(RESP_LINKSTAT_SPEED, lstat)];
 	linfo->an = FIELD_GET(RESP_LINKSTAT_AN, lstat);
 	linfo->fec = FIELD_GET(RESP_LINKSTAT_FEC, lstat);
-	linfo->port = FIELD_GET(RESP_LINKSTAT_PORT, lstat);
 	linfo->lmac_type_id = cgx_get_lmac_type(cgx, lmac_id);
 	lmac_string = cgx_lmactype_string[linfo->lmac_type_id];
 	strncpy(linfo->lmac_type, lmac_string, LMACTYPE_STR_LEN - 1);
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx_fw_if.h b/drivers/net/ethernet/marvell/octeontx2/af/cgx_fw_if.h
index f59882ba6837..4935f7dbfed8 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx_fw_if.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx_fw_if.h
@@ -228,8 +228,9 @@ struct cgx_lnk_sts {
 	uint64_t err_type:10;
 	uint64_t an:1;			/* AN supported or not */
 	uint64_t fec:2;			/* FEC type if enabled, if not 0 */
-	uint64_t port:8;
-	uint64_t reserved2:28;
+	uint64_t lmac_type:8;
+	uint64_t mode:8;
+	uint64_t reserved2:20;
 };
 
 #define RESP_LINKSTAT_UP		GENMASK_ULL(9, 9)
@@ -238,7 +239,7 @@ struct cgx_lnk_sts {
 #define RESP_LINKSTAT_ERRTYPE		GENMASK_ULL(24, 15)
 #define RESP_LINKSTAT_AN		GENMASK_ULL(25, 25)
 #define RESP_LINKSTAT_FEC		GENMASK_ULL(27, 26)
-#define RESP_LINKSTAT_PORT		GENMASK_ULL(35, 28)
+#define RESP_LINKSTAT_LMAC_TYPE		GENMASK_ULL(35, 28)
 
 /* scratchx(1) CSR used for non-secure SW->ATF communication
  * This CSR acts as a command register
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
index 6617d4f52616..45c5481666a4 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
@@ -515,7 +515,6 @@ struct cgx_link_user_info {
 	uint64_t speed:20; /* speed in Mbps */
 	uint64_t an:1;	  /* AN supported or not */
 	uint64_t fec:2;	 /* FEC type if enabled else 0 */
-	uint64_t port:8;
 #define LMACTYPE_STR_LEN 16
 	char lmac_type[LMACTYPE_STR_LEN];
 };
-- 
2.31.1

